code/game/turfs/simulated/floor_damage.dm

```diff
@@ -50,8 +50,11 @@
 	if (health <= 0)
 		//Leftover damage will carry over to whatever tile replaces this one
 		var/leftover = abs(health)
+		//Small sanity to prevent -inf, also prevents recusively destorying tiles that (note that if floor tiles ever get less then 100 HP)
+		if(leftover >= 100)
+			leftover = 99
 		make_plating() //Destroy us and make the plating underneath
-		spawn()
+		spawn(0)
```

Comment:
This change will slow the loop down, but if there's a fuckton of damage it'll also mean that the tile will break over the span of a second or two instead of instantly.

I'm not sure if this change is as good as I thought when I mentioned it. Might be best to remove this.